actioncable/app/javascript/action_cable/index.js

```diff
@@ -30,14 +30,16 @@ export function getConfig(name) {
 }
 
 export function createWebSocketURL(url) {
-  if (url && !/^wss?:/i.test(url)) {
+  const webSocketURL = typeof url === 'function' ? url() : url;
```

Comment:
@kaspth @audiolion I'm not quite following how this change addresses the original issue (https://github.com/rails/rails/issues/33064)

Here, `url` (if it is a function) is being invoked immediately when `createWebSocketURL` is invoked. Thus, calling:
```js
createWebSocketURL(generateURL)
```
with these changes is equivalent to
```js
createWebSocketURL(generateURL())
```
prior to this PR, right?

As I understand it, the original issue was having the created `Consumer`'s `url` property be computed dynamically. In other words, if this PR addressed that issue, then I would expect the new test to pass if it were modified with this change:

```diff
diff --git a/actioncable/test/javascript/src/unit/action_cable_test.js b/actioncable/test/javascript/src/unit/action_cable_test.js
index c9d34abc6d..2b89439484 100644
--- a/actioncable/test/javascript/src/unit/action_cable_test.js
+++ b/actioncable/test/javascript/src/unit/action_cable_test.js
@@ -43,11 +43,16 @@ module("ActionCable", () => {
     })
 
     test("uses function to generate URL", assert => {
+      let dynamicURL = testURL
+
       const generateURL = () => {
-        return testURL
+        return dynamicURL
       }
       const consumer = ActionCable.createConsumer(generateURL)
       assert.equal(consumer.url, testURL)
+
+      dynamicURL = `${testURL}foo`
+      assert.equal(consumer.url, `${testURL}foo`)
     })
   })
 })
```

However, it fails:
```
Expected: "ws://cable.example.com/foo"
Actual: "ws://cable.example.com/"
```

Am I missing something?

(cc @javan)

Reply:
That sounds right, @rmacklin. For this to work correctly, we should always pass a function to `Consumer`, not a string, and the consumer should call that function whenever it opens a new connection.

Reply:
great point @rmacklin let me correct this.